Src/Newtonsoft.Json/Utilities/JavaScriptUtils.cs

```diff
@@ -314,5 +313,175 @@ public static string ToEscapedJavaScriptString(string value, char delimiter, boo
                 return w.ToString();
             }
         }
+
+#if !(NET20 || NET35 || NET40 || PORTABLE40)
+
+        public static async Task<char[]> WriteEscapedJavaScriptStringAsync(TextWriter writer, string s, char delimiter, bool appendDelimiters,
+    bool[] charEscapeFlags, StringEscapeHandling stringEscapeHandling, IArrayPool<char> bufferPool, char[] writeBuffer, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            // leading delimiter
+            if (appendDelimiters)
+            {
+                await writer.WriteAsync(delimiter, cancellationToken).ConfigureAwait(false);
+            }
+            else
+            {
+                cancellationToken.ThrowIfCancellationRequested();
+            }
+
+            if (s != null)
+            {
+                int lastWritePosition = 0;
+
+                for (int i = 0; i < s.Length; i++)
+                {
+                    char c = s[i];
+
+                    if (c < charEscapeFlags.Length && !charEscapeFlags[c])
+                    {
+                        continue;
+                    }
+
+                    string escapedValue;
+
+                    switch (c)
+                    {
+                        case '\t':
+                            escapedValue = @"\t";
+                            break;
+                        case '\n':
+                            escapedValue = @"\n";
+                            break;
+                        case '\r':
+                            escapedValue = @"\r";
+                            break;
+                        case '\f':
+                            escapedValue = @"\f";
+                            break;
+                        case '\b':
+                            escapedValue = @"\b";
+                            break;
+                        case '\\':
+                            escapedValue = @"\\";
+                            break;
+                        case '\u0085': // Next Line
+                            escapedValue = @"\u0085";
+                            break;
+                        case '\u2028': // Line Separator
+                            escapedValue = @"\u2028";
+                            break;
+                        case '\u2029': // Paragraph Separator
+                            escapedValue = @"\u2029";
+                            break;
+                        default:
+                            if (c < charEscapeFlags.Length || stringEscapeHandling == StringEscapeHandling.EscapeNonAscii)
+                            {
+                                if (c == '\'' && stringEscapeHandling != StringEscapeHandling.EscapeHtml)
```

Comment:
Yes, it'd be fine for us English-speakers with our 99% ASCII range script, but I'll have Russian and Japanese coders processing text-heavy JSON cursing my name 😉